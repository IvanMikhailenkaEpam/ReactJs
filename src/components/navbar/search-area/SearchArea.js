import React, { Component } from 'react';
import './SearchArea.css';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setSearchBy, setSearchValue } from '../../../store/movie-search/actions';

class SearchArea extends Component {
  static propTypes = {
    searchValue: PropTypes.string.isRequired,
    searchBy: PropTypes.string.isRequired,
    setSearchValueConnect: PropTypes.func.isRequired,
    setSearchByConnect: PropTypes.func.isRequired,
  };

  constructor(state) {
    super(state);
    const {
      searchValue, searchBy, setSearchValueConnect, setSearchByConnect,
    } = this.props;
    setSearchValueConnect(searchValue || '');
    setSearchByConnect(searchBy || 'title');
  }

  getSearchURL() {
    const { searchValue, searchBy } = this.props;
    if (!searchValue || searchValue === '') {
      return '/';
    }
    return `/search?searchBy=${searchBy}&searchValue=${
      searchValue}`;
  }

  setSearchValue(text) {
    const { setSearchValueConnect } = this.props;
    setSearchValueConnect(text);
  }

  searchByParamClick(event, param) {
    const { setSearchByConnect } = this.props;
    setSearchByConnect(param);
  }

  render() {
    const { searchValue, searchBy } = this.props;
    return (
      <div className="row mt-1 w-100">
        <div className="col">
          <span className="pink font-weight-bold">netflixroulette</span>
        </div>
        <div className="col-12 mt-5 white font-weight-bold"><span>FIND YOUR MOVIE</span></div>
        <div className="col-12">
          <form className="col-12 form-inline my-2 my-lg-0 pl-0">
            <input
              className="form-control mr-sm-2 white search-aria"
              type="text"
              placeholder="Search"
              aria-label="Search"
              value={searchValue || ''}
              onChange={event => this.setSearchValue(event.target.value)}
            />
          </form>
        </div>

        <div className="col-4 mt-3 mb-4 ">
          <span className="mr-3 white font-weight-bold">SEARCH BY</span>
          <button
            id="titleBtn"
            className={`btn my-2 my-sm-0 white ${searchBy === 'title'
              ? 'button-pink' : 'button-black'}`}
            type="button"
            onClick={event => this.searchByParamClick(event, 'title')}
          >
            TITLE
          </button>
          <button
            id="genreBtn"
            className={`btn my-2 ml-2 my-sm-0 white ${searchBy === 'genres'
              ? 'button-pink' : 'button-black'}`}
            type="button"
            onClick={event => this.searchByParamClick(event, 'genres')}
          >
            GENRE
          </button>
        </div>
        <div className="col-2 mt-3 mb-4 ml-auto search-button">
          <Link to={this.getSearchURL()}>
            <button
              className="btn my-2 my-sm-0 white button-pink"
              type="button"
            >
              SEARCH
            </button>
          </Link>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  searchValue: state.movieSearch.searchValue,
  searchBy: state.movieSearch.searchBy,
});
const mapDispatchToProps = {
  setSearchValueConnect: setSearchValue,
  setSearchByConnect: setSearchBy,
};


export default connect(mapStateToProps, mapDispatchToProps)(SearchArea);
