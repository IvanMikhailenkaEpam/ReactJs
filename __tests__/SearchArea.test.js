import React from 'react';
import { shallow } from 'enzyme';
import SearchArea from "../src/components/navbar/search-area/SearchArea";

describe('SearchArea', () => {
  it('renders SearchArea correctly', () => {
    const component = shallow(<SearchArea />);

    expect(component).toMatchSnapshot();
  });

  it('titleBtn click should mach snapshot', () => {
    const component = shallow(<SearchArea />);

    component.find('button#titleBtn').simulate('click');

    expect(component).toMatchSnapshot();
  });

  it('genreBtn click should mach snapshot', () => {
    const component = shallow(<SearchArea />);

    component.find('button#genreBtn').simulate('click');

    expect(component).toMatchSnapshot();
  });

  it('setSearchText function has been called', () => {
    const setSearchTextMock = jest.spyOn(SearchArea.prototype, 'setSearchText');
    const event = {target: {value: "someText"}};
    const component = shallow(<SearchArea />);
    component.find('input').simulate('change', event);

    expect(setSearchTextMock).toHaveBeenCalled();
  });
});
