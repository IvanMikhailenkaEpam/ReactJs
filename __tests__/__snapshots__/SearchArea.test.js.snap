// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchArea genreBtn click should mach snapshot 1`] = `
<div
  className="row mt-1 w-100"
>
  <div
    className="col"
  >
    <span
      className="pink font-weight-bold"
    >
      netflixroulette
    </span>
  </div>
  <div
    className="col-12 mt-5 white font-weight-bold"
  >
    <span>
      FIND YOUR MOVIE
    </span>
  </div>
  <div
    className="col-12"
  >
    <form
      className="col-12 form-inline my-2 my-lg-0 pl-0"
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2 white search-aria"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    className="col-4 mt-3 mb-4 "
  >
    <span
      className="mr-3 white font-weight-bold"
    >
      SEARCH BY
    </span>
    <button
      className="btn my-2 my-sm-0 white button-black"
      id="titleBtn"
      onClick={[Function]}
      type="button"
    >
      TITLE
    </button>
    <button
      className="btn my-2 ml-2 my-sm-0 white button-pink"
      id="genreBtn"
      onClick={[Function]}
      type="button"
    >
      GENRE
    </button>
  </div>
  <div
    className="col-2 mt-3 mb-4 ml-auto search-button"
  >
    <Link
      to="/"
    >
      <button
        className="btn my-2 my-sm-0 white button-pink"
        type="button"
      >
        SEARCH
      </button>
    </Link>
  </div>
</div>
`;

exports[`SearchArea genreBtn click should mach snapshot 2`] = `
<div
  className="row mt-1 w-100"
>
  <div
    className="col"
  >
    <span
      className="pink font-weight-bold"
    >
      netflixroulette
    </span>
  </div>
  <div
    className="col-12 mt-5 white font-weight-bold"
  >
    <span>
      FIND YOUR MOVIE
    </span>
  </div>
  <div
    className="col-12"
  >
    <form
      className="col-12 form-inline my-2 my-lg-0 pl-0"
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2 white search-aria"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    className="col-4 mt-3 mb-4 "
  >
    <span
      className="mr-3 white font-weight-bold"
    >
      SEARCH BY
    </span>
    <button
      className="btn my-2 my-sm-0 white button-black"
      id="titleBtn"
      onClick={[Function]}
      type="button"
    >
      TITLE
    </button>
    <button
      className="btn my-2 ml-2 my-sm-0 white button-pink"
      id="genreBtn"
      onClick={[Function]}
      type="button"
    >
      GENRE
    </button>
  </div>
  <div
    className="col-2 mt-3 mb-4 ml-auto search-button"
  >
    <Link
      to="/"
    >
      <button
        className="btn my-2 my-sm-0 white button-pink"
        type="button"
      >
        SEARCH
      </button>
    </Link>
  </div>
</div>
`;

exports[`SearchArea renders SearchArea correctly 1`] = `
<div
  className="row mt-1 w-100"
>
  <div
    className="col"
  >
    <span
      className="pink font-weight-bold"
    >
      netflixroulette
    </span>
  </div>
  <div
    className="col-12 mt-5 white font-weight-bold"
  >
    <span>
      FIND YOUR MOVIE
    </span>
  </div>
  <div
    className="col-12"
  >
    <form
      className="col-12 form-inline my-2 my-lg-0 pl-0"
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2 white search-aria"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    className="col-4 mt-3 mb-4 "
  >
    <span
      className="mr-3 white font-weight-bold"
    >
      SEARCH BY
    </span>
    <button
      className="btn my-2 my-sm-0 white button-pink"
      id="titleBtn"
      onClick={[Function]}
      type="button"
    >
      TITLE
    </button>
    <button
      className="btn my-2 ml-2 my-sm-0 white button-black"
      id="genreBtn"
      onClick={[Function]}
      type="button"
    >
      GENRE
    </button>
  </div>
  <div
    className="col-2 mt-3 mb-4 ml-auto search-button"
  >
    <Link
      to="/"
    >
      <button
        className="btn my-2 my-sm-0 white button-pink"
        type="button"
      >
        SEARCH
      </button>
    </Link>
  </div>
</div>
`;

exports[`SearchArea titleBtn click should mach snapshot 1`] = `
<div
  className="row mt-1 w-100"
>
  <div
    className="col"
  >
    <span
      className="pink font-weight-bold"
    >
      netflixroulette
    </span>
  </div>
  <div
    className="col-12 mt-5 white font-weight-bold"
  >
    <span>
      FIND YOUR MOVIE
    </span>
  </div>
  <div
    className="col-12"
  >
    <form
      className="col-12 form-inline my-2 my-lg-0 pl-0"
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2 white search-aria"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    className="col-4 mt-3 mb-4 "
  >
    <span
      className="mr-3 white font-weight-bold"
    >
      SEARCH BY
    </span>
    <button
      className="btn my-2 my-sm-0 white button-pink"
      id="titleBtn"
      onClick={[Function]}
      type="button"
    >
      TITLE
    </button>
    <button
      className="btn my-2 ml-2 my-sm-0 white button-black"
      id="genreBtn"
      onClick={[Function]}
      type="button"
    >
      GENRE
    </button>
  </div>
  <div
    className="col-2 mt-3 mb-4 ml-auto search-button"
  >
    <Link
      to="/"
    >
      <button
        className="btn my-2 my-sm-0 white button-pink"
        type="button"
      >
        SEARCH
      </button>
    </Link>
  </div>
</div>
`;
